src/Controller/Controller.php

```diff
@@ -496,6 +496,10 @@ public function startupProcess()
         if ($event->result instanceof Response) {
             return $event->result;
         }
+        $event = $this->dispatchEvent('Controller.beforeFilter');
+        if ($event->result instanceof Response) {
+            return $event->result;
+        }
```

Comment:
I think this event is in the wrong spot. The beforeFilter methods should all be invoked before startup.